src/wallet/wallet.cpp

```diff
@@ -924,6 +924,45 @@ void CWallet::LoadToWallet(CWalletTx& wtxIn)
     }
 }
 
+void CWallet::RemoveOldTransactions(const CScript& script)
```

Comment:
I think `RemoveTransactions()` would be a better name, as this can remove any transactions, being "old" is not a requirement.